xla/service/buffer_assignment.h

```diff
@@ -570,7 +573,8 @@ class BufferAssignment {
 
   // Combines allocations of temporary buffers into one big BufferAllocation.
   void CombineTempAllocations(
-      const absl::flat_hash_set<BufferValue::Color>& private_stack_colors);
+      const absl::flat_hash_set<BufferValue::Color>& private_stack_colors,
+      bool enable_temp_user_buffer);
```

Comment:
Is this a GPU-specific option? Should it go into a GPU subclass?

Reply:
Currently this feature is only valid for CUDA devices. but in theory it should also work for ROCM or other devices, as long as we use a separate memory allocator for temp buffer. BufferAssignment pass is a general optimization pass, so currently I made a patch here.